src/fptostring.cpp

```diff
@@ -0,0 +1,208 @@
+// SPDX-FileCopyrightText: 2024 Simon Gene Gottlieb
+// SPDX-License-Identifier: MIT
+
+#include "yaml-cpp/fptostring.h"
+#include "contrib/dragonbox.h"
+
+#include <array>
+#include <cassert>
+#include <cmath>
+#include <limits>
+#include <sstream>
+#include <tuple>
+
+namespace YAML {
+namespace detail {
+namespace fp_formatting {
+
+/**
+ * Converts a integer into its ASCII digits.
+ *
+ * @param begin/end - a buffer, must be at least 20bytes long
+ * @param value   - input value
+ * @param width   - minimum number of digits, fill with '0' to the left. Must be equal or smaller than the buffer size.
+ * @return      - number of digits filled into the buffer.
+ *
+ * Example:
+ * std::array<char, 20> buffer;
+ * auto ct = ConvertToChars(buffer.begin(), buffer.end(), 23, 3);
+ * assert(ct = 3);
+ * assert(buffer[0] == '0');
+ * assert(buffer[1] == '2');
+ * assert(buffer[2] == '3');
+ */
+int ConvertToChars(char* begin, char* end, size_t value, int width=1) {
+  assert(width >= 1);
+  assert(end >= begin);     // end must be after begin
+  assert(end-begin >= width); // Buffer must be large enough
```

Comment:
These asserts should be early returns also, at a minimum - if not, this could be a buffer overrun in prod.

Reply:
Preferred exception you want me to throw?

Reply:
I would say: assert, then return -1; document that negative return values mean errors, and check for that in the call site.

Reply:
Changed it, and put an fallback to std::stringstream intoplace.